### ISSUE TITLE
tools.use_downloader should print error to stdout if curl command fails

### ISSUE THREAD

**Issue by justinmk** [luarocks/luarocks#283]
stderr appears to be discarded for curl:

https://github.com/keplerproject/luarocks/blob/3abde0dc4f1c3a246026d01e871ca9d4c165c57d/src/luarocks/fs/unix/tools.lua#L269

If I recall, this is because `curl` sends HTTP headers to stderr. But this has the side effect that if curl fails, user doesn't know why. Perhaps stderr could be redirected to a file, and if curl fails, then the file could be printed to stdout?

We ran into this here: https://github.com/neovim/neovim/issues/1170#issuecomment-55588635
 To summarize: the user's `~/.cache/luarocks/https___rocks.moonscript.org/manifest-5.1` directory was inaccessible, which caused curl to fail.

related: https://github.com/keplerproject/luarocks/issues/11

**Comment by justinmk** [comment]
Haven't seen related problem in a long time, maybe this can be closed.
